006_team_4_agent_design/006_team_4_agent_design.tex

```diff
@@ -1 +1,311 @@
+
+In the cases where the floor above left a certain amount of food and the agent below also took a certain amount of food, the agent did not instantaneously increase the trust score. Instead, the agent verified that the agents above and below followed through with the actions mentioned through the \texttt{verifyResponses} function. The agent used the function once it received replies to \texttt{RequestLeaveFood} and \texttt{RequestTakeFood} messages. Whenever a message was sent, it was stored by the agent. This allowed the agent to correspond replies and responses to the appropriate sent messages. All responses which were not in accordance with our agent's ideology, or were not replied to at all, were sanctioned through a decrease in the global trust score metric. A decrease in trust score led the agent to act more selfishly, which acted to the detriment of other agents. This meant that there was some form of punishment for antisocial behaviour from other agents to stop them destroying the social order constructed by the agent within the tower.
+
+In the case of the agent on the above floor leaving food, the agent first checked whether the platform was on its current floor through the function \texttt{PlatformOnFloor}. The global trust score was then updated through the function \texttt{updateGlobalTrustLeaveFood}. The agent used the function \texttt{CurrPlatFood} to check the food on the platform and verify that the food on the platform was greater than or equal to the food that the agent on the above floor had said they were going to leave. This resulted in an increase in the trust score. In the case where an agent lied (though this should not be possible in simulation), the agent reduced its trust score as a consequence of lying. 
+
+In the case of the agent below taking food, the agent always stored the food left on the platform after they had eaten. When a response to a \texttt{RequestTakeFood} message was received from the below floor, the agent first checked whether the below agent had eaten through the function \texttt{neighbourFoodEaten}. As the food left on the platform was stored by the agent and the agent being able to see the food on the below platform through the function \texttt{CurrPlatFood}, the agent was able to verify that the agent below had been truthful. This was done by adding the food that the agent had supposedly taken to the food that was currently on the platform and verifying it against the food that had left this agent's platform.
+
+We updated the trust score based on the messages sent and received to and from other agents. This showcased self optimisation, since the agent chose a configuration to follow based on interactions with others and observation of their actions. We then used this trust score as an indicator of when to switch between our three different personalities. With high trust, our agent was inclined to be more selfless, and with low trust our agent was likely to be more selfish.
```

Comment:
```suggestion
The agent updated the trust score based on the messages sent and received to and from other agents. This showcased self optimisation, since the agent chose a configuration to follow based on interactions with others and observation of their actions. The agent then used this trust score as an indicator of when to switch between the three different personalities. With high trust, the agent was inclined to be more selfless, and with low trust the agent was likely to be more selfish.
```